### PR TITLE
Epic Seedless-onboarding

### LINKED ISSUE #2452: Epic: [Seedless-Onboarding]
## What is the feature about
We want to integrate a solution for seedless onboarding by logging in through an social account.

This is the epic issue. Separate issues will be created for each requirement / user story.

## The list of requirements
- [ ] Integrate authenticators
  - [x] Google
  - [ ] (optional) #2457 v2
- [x] Device share
  - [x] Store device share in local storage
- [ ] Recovery
  - [x] #2455 
  - [ ] (optional) #2486 v2
  - [x] Build Single factor recovery
- [x] Wallet integration
  - [x] #2456
- [x] Blockchain interactions (are done via Provider, should work once the Web3Provider can be created)
  - The returned provider does not correctly implement `eth_estimateGas`, we had to patch it's `eth_estimateGas` method to call the readonly provider's `eth_estimateGas` method. 
  - [x] Sign messages
  - [x] Sign transactions
- [ ] Implement designs / user flows
  - [x] #2459
  - [x] #2569
  - [x] New quick Safe creation flow
  - [x] Update Account center
  - [ ] (optional) Add global MFA prompt
  - [x] #2589
  - [x] #2519
- [x] #2458

## Designs/sketches
tbd

## Links
https://www.notion.so/safe-global/Feature-spec-d4b73908e0804f22b9c92565ec8f98c0

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves
Implements seedless onboarding via MPC Core SDK

Resolves #2452 

## How to test

- Login with Google via Header
- Continue with Google via Welcome page
- Set password
- Extract pk
- Recover with password
- Recover with password and remember

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/hooks/wallets/mpc/useMPC.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      chainId: `0x${Number(chain.chainId).toString(16)}`,
+      chainNamespace: CHAIN_NAMESPACES.EIP155,
+      rpcTarget: getRpcServiceUrl(chain.rpcUri),
+      displayName: chain.chainName,
+      blockExplorer: chain.blockExplorerUriTemplate.address,
+      ticker: chain.nativeCurrency.symbol,
+      tickerName: chain.nativeCurrency.name,
+    }
+
+    const currentInstance = getStore()
```
Comment (iamacook) [inline src/hooks/wallets/mpc/useMPC.ts:32]: `getStore` is not reactive. We should access `useStore` instead.

File: src/hooks/wallets/mpc/useMPC.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            )
+            newProvider.emit('chainChanged', `0x${Number(chainConfig.chainId).toString(16)}`)
+          }
+        }
+      })
+      .catch((error) => console.error(error))
+  }, [chain, onboard])
+}
+
+export const getMPCCoreKitInstance = getStore
```
Comment (iamacook) [inline src/hooks/wallets/mpc/useMPC.ts:82]: Nit: let's export this prefixed as per my above comment. I see that this is only used inside the onboard module, which I _think_ is alright but it shouldn't used within React.

```suggestion
export const _getMPCCoreKitInstance = getStore
```

**Review by iamacook — COMMENTED** [review]
Please note: this is not a full review.

These are findings from reviewing #2559 that are out of scope there.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ concurrency:
 
 jobs:
   eslint:
+    permissions: write-all
```
Comment (katspaugh) [inline .github/workflows/lint.yml]: Rm this please

Comment (usame-algan) [reply .github/workflows/lint.yml]: Missed it 😓 Its removed now

File: .github/workflows/build/action.yml [hunk 4]
```diff
@@ -43,6 +43,10 @@ runs:
         NEXT_PUBLIC_SAFE_RELAY_SERVICE_URL_STAGING: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_SAFE_GELATO_RELAY_SERVICE_URL_STAGING }}
         NEXT_PUBLIC_IS_OFFICIAL_HOST: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_IS_OFFICIAL_HOST }}
         NEXT_PUBLIC_REDEFINE_API: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_REDEFINE_API }}
+        NEXT_PUBLIC_WEB3AUTH_CLIENT_ID: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_WEB3AUTH_CLIENT_ID }}
+        NEXT_PUBLIC_WEB3AUTH_AGGREGATE_VERIFIER_ID: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_WEB3AUTH_AGGREGATE_VERIFIER_ID }}
+        NEXT_PUBLIC_WEB3AUTH_SUBVERIFIER_ID: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_WEB3AUTH_SUBVERIFIER_ID }}
+        NEXT_PUBLIC_GOOGLE_CLIENT_ID: ${{ fromJSON(inputs.secrets).NEXT_PUBLIC_GOOGLE_CLIENT_ID }}
```
Comment (katspaugh) [inline .github/workflows/build/action.yml:49]: This needs to be updated, right?

File: src/config/constants.ts [hunk 5]
```diff
@@ -98,3 +98,9 @@ export const IS_OFFICIAL_HOST = process.env.NEXT_PUBLIC_IS_OFFICIAL_HOST === 'tr
 export const REDEFINE_SIMULATION_URL = 'https://dashboard.redefine.net/reports/'
 export const REDEFINE_API = process.env.NEXT_PUBLIC_REDEFINE_API
 export const REDEFINE_ARTICLE = 'https://safe.mirror.xyz/rInLWZwD_sf7enjoFerj6FIzCYmVMGrrV8Nhg4THdwI'
+
+// Social Login
+export const WEB3_AUTH_CLIENT_ID = process.env.NEXT_PUBLIC_WEB3AUTH_CLIENT_ID || ''
+export const GOOGLE_CLIENT_ID = process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID || ''
+export const WEB3AUTH_SUBVERIFIER_ID = process.env.NEXT_PUBLIC_WEB3AUTH_SUBVERIFIER_ID || ''
+export const WEB3AUTH_AGGREGATE_VERIFIER_ID = process.env.NEXT_PUBLIC_WEB3AUTH_AGGREGATE_VERIFIER_ID || ''
```
Comment (katspaugh) [inline src/config/constants.ts:106]: This should be removed.

**Review by katspaugh — APPROVED** [review]

**Review by usame-algan — APPROVED** [review]

**Review by usame-algan — APPROVED** [review]

**Pull request merged**